tests/next_tests/integration_tests/feature_tests/ffront_tests/test_program.py

```diff
@@ -36,205 +40,260 @@
+    @gtx.program
     def prog(
-        a: Field[[IDim], float64],
-        b: Field[[IDim], float64],
-        out_a: Field[[IDim], float64],
-        out_b: Field[[IDim], float64],
+        a: cases.IFloatField,
+        b: cases.IFloatField,
+        out_a: cases.IFloatField,
+        out_b: cases.IFloatField,
     ):
         pack_tuple(a, b, out=(out_a, out_b))
 
-    prog(a, b, out_a, out_b, offset_provider={})
-
-    assert np.allclose(a, out_a)
-    assert np.allclose(b, out_b)
-
+    a = cases.allocate(cartesian_case, prog, "a")()
+    b = cases.allocate(cartesian_case, prog, "b")()
+    out_a = cases.allocate(cartesian_case, prog, "out_a")()
+    out_b = cases.allocate(cartesian_case, prog, "out_b")()
+
+    cases.verify(
+        cartesian_case,
+        prog,
+        a,
+        b,
+        out_a,
+        out_b,
+        inout=(out_a.array(), out_b.array()),
+        ref=(a.array(), b.array()),
+        comparison=lambda a, b: (np.all(a == out_a), np.all(b == out_b)),
+    )
 
-def test_tuple_program_return_constructed_inside_with_slicing(fieldview_backend):
-    size = 10
-    a = np_as_located_field(IDim)(np.ones((size,)))
-    b = np_as_located_field(IDim)(2 * np.ones((size,)))
-    out_a = np_as_located_field(IDim)(np.zeros((size,)))
-    out_b = np_as_located_field(IDim)(np.zeros((size,)))
 
-    @field_operator
+def test_tuple_program_return_constructed_inside_with_slicing(cartesian_case):
+    @gtx.field_operator
     def pack_tuple(
-        a: Field[[IDim], float64], b: Field[[IDim], float64]
-    ) -> tuple[Field[[IDim], float64], Field[[IDim], float64]]:
+        a: cases.IFloatField, b: cases.IFloatField
+    ) -> tuple[cases.IFloatField, cases.IFloatField]:
         return (a, b)
 
-    @program(backend=fieldview_backend)
+    @gtx.program
     def prog(
-        a: Field[[IDim], float64],
-        b: Field[[IDim], float64],
-        out_a: Field[[IDim], float64],
-        out_b: Field[[IDim], float64],
+        a: cases.IFloatField,
+        b: cases.IFloatField,
+        out_a: cases.IFloatField,
+        out_b: cases.IFloatField,
     ):
         pack_tuple(a, b, out=(out_a[1:], out_b[1:]))
 
-    prog(a, b, out_a, out_b, offset_provider={})
-
-    assert np.allclose(a[1:], out_a[1:])
-    assert out_a[0] == 0.0
-    assert np.allclose(b[1:], out_b[1:])
-    assert out_b[0] == 0.0
-
+    a = cases.allocate(cartesian_case, prog, "a").strategy(cases.ConstInitializer(1))()
+    b = cases.allocate(cartesian_case, prog, "b").strategy(cases.ConstInitializer(2))()
+    out_a = cases.allocate(cartesian_case, prog, "out_a").strategy(cases.ConstInitializer(0))()
+    out_b = cases.allocate(cartesian_case, prog, "out_b").strategy(cases.ConstInitializer(0))()
+
+    cases.verify(
+        cartesian_case,
+        prog,
+        a,
+        b,
+        out_a,
+        out_b,
+        inout=(out_a.array()[1:], out_b.array()[1:]),
+        ref=(a.array()[1:], b.array()[1:]),
+        comparison=lambda a, b: (
```

Comment:
Same issue as below. Non-empty containers are always True in Python. Must return a bool value.

Reply:
The conditions can simply be connected with `and` rather than commas.